### PR TITLE
"DOC #45443 edited the documentation of where/mask functions"

### LINKED ISSUE #45443: BUG: mask with alignment
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd
s = pd.Series(range(5))
t = pd.Series([True, False])
s.mask(t, 99)
```

### Issue Description

Series.mask replaces values if index is missing in condition. Example output:

0    99
1     1
2    99
3    99
4    99
dtype: int64

### Expected Behavior

"Replace values where the condition is True." Values should not be touched if index is missing in condition.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 66e3805b8cabe977f40c05259cc3fcf7ead5687d
python           : 3.8.12.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19042
machine          : AMD64
processor        : Intel64 Family 6 Model 60 Stepping 3, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : de_DE.cp1252

pandas           : 1.3.5
numpy            : 1.22.0
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 21.3.1
setuptools       : 60.2.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : 4.3.2
blosc            : None
feather          : None
xlsxwriter       : 3.0.2
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : 2.9.3 (dt dec pq3 ext lo64)
jinja2           : 3.0.3
IPython          : 7.19.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.5.1
numexpr          : None
odfpy            : None
openpyxl         : 3.0.9
pandas_gbq       : None
pyarrow          : 6.0.1
pyxlsb           : None
s3fs             : None
scipy            : 1.7.3
sqlalchemy       : 1.4.29
tables           : None
tabulate         : 0.8.9
xarray           : None
xlrd             : 2.0.1
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by ahmedibrhm**
- [X] closes #45443
- [X] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -9571,6 +9571,8 @@ def where(
         cond : bool {klass}, array-like, or callable
             Where `cond` is {cond}, keep the original value. Where
             {cond_rev}, replace with corresponding value from `other`.
+            if `cond` {klass} is less in size than `other`, the default bool
```
Comment (mroeschke) [inline pandas/core/generic.py:9574]: I think this would be better demonstrated in the `Example` section below

Comment (ahmedibrhm) [reply pandas/core/generic.py]: @mroeschke What do you think now?

**Review by phofl — APPROVED** [review]
lgtm

File: pandas/core/generic.py [hunk 2]
```diff
@@ -9641,6 +9642,23 @@ def where(
         4    NaN
         dtype: float64
 
+        >>> s = pd.Series(range(5))
+        >>> t = pd.Series([True, False])
+        >>> s.where(t,99)
```
Comment (mroeschke) [inline pandas/core/generic.py:9647]:
```suggestion
        >>> s.where(t, 99)
```

File: pandas/core/generic.py [hunk 3]
```diff
@@ -9614,7 +9614,8 @@ def where(
         The {name} method is an application of the if-then idiom. For each
         element in the calling DataFrame, if ``cond`` is ``{cond}`` the
         element is used; otherwise the corresponding element from the DataFrame
-        ``other`` is used.
+        ``other`` is used. If `cond` {klass} is less in size than `other`, the default bool
```
[the remark this answers is not in the document]
Comment (ahmedibrhm) [reply pandas/core/generic.py]: @mroeschke done

[the remark this answers is not in the document]
Comment (mroeschke) [reply pandas/core/generic.py]: Looks like the validation checks will still fail: https://github.com/pandas-dev/pandas/pull/47626#issuecomment-1178008466=

Please review how to install all the code checks: https://pandas.pydata.org/pandas-docs/stable/development/contributing_codebase.html#pre-commit

[the remark this answers is not in the document]
Comment (ahmedibrhm) [reply pandas/core/generic.py]: @mroeschke It is ok now. Sorry for the inconvenience!

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks for the responsive updates @ahmedibrhm

**Pull request merged**